client/main.lua

```diff
@@ -20,34 +20,44 @@ end
 lib.callback.register('qbx_truckrobbery:resetMission', resetMission)
 
 local function lootTruck()
-	if lib.progressBar({
-		duration = config.lootDuration,
-		label = locale('info.looting_truck'),
-		useWhileDead = false,
-		canCancel = true,
-		disable = {
-			move = true,
-			car = true,
-			combat = true,
-			mouse = false,
-		},
-		anim = {
-			dict = 'anim@heists@ornate_bank@grab_cash_heels',
-			clip = 'grab',
-			flag = 1,
-		},
-		prop = {
-			model = `prop_cs_heist_bag_02`,
-			bone = 57005,
-			pos = vec3(0.0, 0.0, -0.16),
-			rot = vec3(250.0, -30.0, 0.0),
-		}
-	}) then
-		local success = lib.callback.await('qbx_truckrobbery:server:giveReward')
-		if not success then return end
-		SetPedComponentVariation(cache.ped, 5, 45, 0, 2)
-		resetMission()
-	end
+    local looting = true
+	CreateThread(function()
+        if lib.progressBar({
+            duration = config.lootDuration,
+            label = locale('info.looting_truck'),
+            useWhileDead = false,
+            canCancel = true,
+            disable = {
+                move = true,
+                car = true,
+                combat = true,
+                mouse = false,
+            },
+            anim = {
+                dict = 'anim@heists@ornate_bank@grab_cash_heels',
+                clip = 'grab',
+                flag = 1,
+            },
+            prop = {
+                model = `prop_cs_heist_bag_02`,
+                bone = 57005,
+                pos = vec3(0.0, 0.0, -0.16),
+                rot = vec3(250.0, -30.0, 0.0),
+            }
+        }) then
+            local success = lib.callback.await('qbx_truckrobbery:server:giveReward')
+            if not success then return end
+            SetPedComponentVariation(cache.ped, 5, 45, 0, 2)
+            resetMission()
+        end
+        looting = false
+    end)
+    while looting do
+        if #(GetEntityCoords(cache.ped) - GetEntityCoords(truck)) > 3 then
+            lib.cancelProgress()
+        end
+        Wait(1000)
+    end
```

Comment:
That might cause a race condition, no? The progress may not be active yet at the time the while loop starts since it's been started in a different thread.

Reply:
How I normally do it is like this

```lua
lib.callback.register('Renewed-Vehiclekeys:client:stealFromPed', function(netId)
    SetTimeout(100, function()
        while lib.progressActive() do

        end
    end)

    return lib.progressCircle({
        duration = 3000,
        label = locale('taking_keys'),
        position = 'bottom',
    })
end)
```

Reply:
I prefer the boolean way than using a timeout. It's less error prone. Although the timeout will likely work, we cannot guarantee that code is executed within a given timeframe, and so such solutions can be brittle.